### PR TITLE
build: upgrade to google.golang.org/protobuf

### PR DESCRIPTION
**Pull request by michaelbeaumont**
In preparation for #2719 

Blocked by #4830 

Note that we continue to use `github.com/golang/protobuf/jsonpb` [due to the old library unmarshaling go-like `time.Duration`s even though it's against protobuf spec](https://github.com/golang/protobuf/issues/1374).

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests -- none
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](/CONTRIBUTING.md#backporting)? --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/mads/v1/service/http.go [hunk 1]
```diff
@@ -84,14 +86,15 @@ func (s *service) handleDiscovery(req *restful.Request, res *restful.Response) {
 		return
 	}
 
-	marshaller := &jsonpb.Marshaler{OrigName: true}
-	resStr, err := marshaller.MarshalToString(discoveryRes)
+	// TODO is this equivalent?
```
Comment (jakubdyszkiewicz) [inline pkg/mads/v1/service/http.go:89]: Is it?

**Review by jakubdyszkiewicz — APPROVED** [review]
great work!!

File: pkg/mads/v1/service/http.go (hunk 1 above)
Comment (michaelbeaumont) [reply pkg/mads/v1/service/http.go:89]: Yes!

https://github.com/golang/protobuf/blob/v1.5.2/jsonpb/encode.go#L99

Comment (michaelbeaumont) [reply pkg/mads/v1/service/http.go:89]: Yes!

https://github.com/golang/protobuf/blob/v1.5.2/jsonpb/encode.go#L99

Comment (michaelbeaumont) [reply pkg/mads/v1/service/http.go]: but actually I'm going to move this back to `jsonpb` for consistency.

**Pull request merged**